lib/ansible/modules/tacp/tacp_network.py

```diff
@@ -0,0 +1,543 @@
+        api_key=dict(type='str', required=True),
+        name=dict(type='str', required=True),
+        state=dict(type='str', default='present',
+                   choices=['present', 'absent']),
+        network_type=dict(type='str', required=True, choices=[
+                          'VLAN', 'vlan', 'VNET', 'vnet']),
+        site_name=dict(type='str', required=False),
+        vlan_tag=dict(type='int', required=False),
+        firewall_override=dict(type='str', required=False),
+        firewall_profile=dict(type='str', required=False),
+        autodeploy_nfv=dict(type='bool', required=False, default=True),
+        nfv=dict(type='dict', required=False),
+        network_address=dict(type='str', required=False),
+        subnet_mask=dict(type='str', required=False),
+        gateway=dict(type='str', required=False),
+        dhcp=dict(type='dict', required=False),
+        routing=dict(type='dict', required=False)
+    )
+
+    # seed the result dict in the object
+    # we primarily care about changed and state
+    # change is if this module effectively modified the target
+    # state will include any data that you want your module to pass back
+    # for consumption, for example, in a subsequent task
+    result = dict(
+        changed=False,
+        args=[]
+    )
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    # if the user is working with this module in only check mode we do not
+    # want to make any changes to the environment, just return the current
+    # state with no modifications
+    if module.check_mode:
+        module.exit_json(**result)
+
+    def fail_with_reason(reason):
+        module.fail_json(**result, msg=reason)
+
+    def get_site_list():
+        api_instance = tacp.LocationsApi(api_client)
+        try:
+            # View sites for an organization
+            api_response = api_instance.get_locations_for_organization_using_get()
+        except ApiException as e:
+            return "Exception when calling get_locations_for_organization_using_get: %s\n" % e
+
+        if api_response:
+            return api_response
+        else:
+            return None
+
+    def validate_input(params):
+        # Check if there is more than one site already
+        # If there is only one, default to that one.
+        # Otherwise require a site_name to be provided
+        site_list = get_site_list()
+        if len(site_list) != 1:
+            required_params.append('site_name')
+
+        inputs_valid = True
+        if params['network_type'].upper() == 'VLAN':
+            if params['state'] == 'present':
+                missing_params = []
+                required_params = ['name', 'vlan_tag']
```

Comment:
It looks like you might be overriding the `site_name` required param here. Also, not sure where `required_params` is defined?

Reply:
Fixed!